src/App.tsx

```diff
@@ -1,26 +1,295 @@
-/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { deleteTodo, getTodos, postTodo, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
+import { TodoStatus } from './types/TodoStatus';
 
-const USER_ID = 0;
+function getFilteredTodos(
+  todos: Todo[],
+  { status }: { status: TodoStatus },
+): Todo[] {
+  let filteredTodos = [...todos];
+
+  if (status === TodoStatus.Active) {
+    filteredTodos = filteredTodos.filter(todo => todo.completed === false);
+  } else if (status === TodoStatus.Completed) {
+    filteredTodos = filteredTodos.filter(todo => todo.completed === true);
+  }
+
+  return filteredTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState(TodoStatus.All);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [loadingTodos, setLoadingTodos] = useState<Todo['id'][]>([]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  function hideErrorMessage() {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
```

Comment:
Why not make one step further?

```suggestion
  function showErrorMessageWithAutoHide(errorText: string) {
    setErrorMessage(errorText);
    // and schedule autoHide
    setTimeout(() => {
      setErrorMessage('');
    }, 3000);
  }
```